voctogui/lib/audioleveldisplay.py

```diff
@@ -129,10 +141,13 @@ def normalize_db(self, db):
     def clamp(self, value, min_value=0, max_value=1):
         return max(min(value, max_value), min_value)
 
-    def level_callback(self, rms, peak, decay):
-        if self.levelrms != rms or self.levelpeak != peak \
-                or self.leveldecay != decay:
-            self.levelrms = rms
-            self.levelpeak = peak
-            self.leveldecay = decay
+    def level_callback(self, rms, peak, decay, stream):
+        p = self.channels * stream
+        if self.levelrms[p] != rms[0] \
```

Comment:
Why is this condition in place? Would be much simpler to just copy the values – and more readable.
Also, comparing floats with == or != is a [no-no](https://floating-point-gui.de/errors/comparison/). Better just skip the comparison completely.

Reply:
These conditions were in the original code, probably to only fire a draw event if the values actually changed.

Reply:
but assigning a value to `self.levelrms` does not in itself trigger a draw event. The event is triggered via the timer.

This is not a criticism of you but of the code, and while you're at it, we might as well make it better :)